keps/sig-scheduling/5055-dra-device-taints-and-tolerations/README.md

```diff
@@ -0,0 +1,768 @@
+    // to NoExecute.
+    //
+    // In addition, should any of the allocated devices get tainted
+    // with NoExecute after allocation and that effect is not tolerated,
+    // then all pods consuming the ResourceClaim get deleted to evict
+    // them. The scheduler will not let new pods reserve the claim while
+    // it has these tainted devices. Once all pods are evicted, the
+    // claim will get deallocated.
+    //
+    // +optional
+    // +listType=atomic
+    Tolerations []Toleration
+}
+
+// The ResourceClaim this Toleration is attached to tolerate any taint that matches
+// the triple <key,value,effect> using the matching operator <operator>.
+type Toleration struct {
+    // Key is the taint key that the toleration applies to. Empty means match all taint keys.
+    // If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+    // Must be a label name.
+    //
+    // +optional
+    Key string
+
+    // Operator represents a key's relationship to the value.
+    // Valid operators are Exists and Equal. Defaults to Equal.
+    // Exists is equivalent to wildcard for value, so that a ResourceClaim can
+    // tolerate all taints of a particular category.
+    //
+    // +optional
+    Operator TolerationOperator
+
+    // Value is the taint value the toleration matches to.
+    // If the operator is Exists, the value should be empty, otherwise just a regular string.
+    // Must be a label value.
+    //
+    // +optional
+    Value string
+
+    // Effect indicates the taint effect to match. Empty means match all taint effects.
+    // When specified, allowed values are NoSchedule and NoExecute.
+    //
+    // +optional
+    Effect TaintEffect
+
+    // TolerationSeconds represents the period of time the toleration (which must be
+    // of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default,
+    // it is not set, which means tolerate the taint forever (do not evict). Zero and
+    // negative values will be treated as 0 (evict immediately) by the system.
+    //
+    // +optional
+    TolerationSeconds *int64
+}
+
+// A toleration operator is the set of operators that can be used in a toleration.
+//
+// +enum
+type TolerationOperator string
+
+const (
+    TolerationOpExists TolerationOperator = "Exists"
+    TolerationOpEqual  TolerationOperator = "Equal"
+)
+```
+
+As with Taint, these structs get duplicated to document DRA specific
+behavior and to ensure that future extensions do not get inherited
+accidentally.
+
+Generated conversion code might make it possible to reuse existing helper
+code. Alternatively, that code can be copied.
+
+The DevicePatch also gets extended. It is possible to use
+admin-controlled taints without enabling attribute overrides by enabling the
+`v1alpha3` API and only the `DRADeviceTaints` feature, while leaving
+`DRAAdminControlledDeviceAttributes` disabled.
```

Comment:
Have a question at here: what's the impact if disabling `DRAAdminControlledDeviceAttributes` while only enabling `DRADeviceTaints` 
And do we need to clarify the behaviour at here

Reply:
Isn't that exactly what I am answering here? :sweat_smile: 

It's supported. Perhaps I need to add a "because then the ResourceSlicePatch type is enabled with only the fields needed for taints". Done.